scripts/workarounds/boot-order/metal-lib.sh

```diff
@@ -0,0 +1,510 @@
+        artifact_error=1
+    fi
+
+    [ "$artifact_error" = 0 ] && return 0 || return 1
+}
+
+function configure_lldp() {
+    local interfaces
+    # Grab all bond interfaces and their members; mgmt and sun NICs. Remove duplicates from partial matches.
+    interfaces=$(ls /sys/class/net/ | grep -oP '(mgmt|bond|sun)\d+' | sort -u)
+    systemctl is-active lldpad >/dev/null || systemctl start lldpad
+    for i in $interfaces; do
+      echo "Configuring LLDP [$i] ..."
+      lldptool set-lldp -i $i adminStatus=rxtx
+      printf '[%s] sysName ' $i && lldptool -T -i $i -V  sysName enableTx=yes
+      printf '[%s] portDesc ' $i && lldptool -T -i $i -V  portDesc enableTx=yes
+      printf '[%s] sysDesc ' $i && lldptool -T -i $i -V  sysDesc enableTx=yes
+      printf '[%s] sysCap ' $i && lldptool -T -i $i -V sysCap enableTx=yes
+      printf '[%s] mngAddr ' $i && lldptool -T -i $i -V mngAddr enableTx=yes
+    done
+    echo 'LLDP is configured for all bond interfaces and their members (mgmt and sun)'
+}
+
+function set_static_fallback() {
+
+    #
+    ## Set static IP; assign the current IP as static.
+    #
+
+    local defgw
+    local ipaddr
+    local lan
+    local netmask
+    local netconf=/tmp/netconf
+
+    # BMCs either run dedicated on lan3 (last LAN channel as is the case with Intel's),
+    # or lan1 (when there's only one channel).
+    if ipmi_output_3=$(ipmitool lan print 3 2>/dev/null); then
+        lan=3
+        echo "$ipmi_output_3" > $netconf
+    elif [ -z $lan ] && ipmi_output_1=$(ipmitool lan print 1 2>/dev/null); then
+        lan=1
+        echo "$ipmi_output_1" > $netconf
+    elif [ -z $lan ]; then
+        echo "Failed to determine which LAN channel to use!"
+    fi
+
+    ipaddr=$(grep -Ei 'IP Address\s+\:' $netconf | awk '{print $NF}')
+    netmask=$(grep -Ei 'Subnet Mask\s+\:' $netconf | awk '{print $NF}')
+    defgw=$(grep -Ei 'Default Gateway IP\s+\:' $netconf | awk '{print $NF}')
+    ipmitool lan set $lan ipsrc static || :
+    ipmitool lan set $lan ipaddr $ipaddr || :
+    ipmitool lan set $lan netmask $netmask || :
+    ipmitool lan set $lan defgw ipaddr $defgw || :
+    ipmitool lan print $lan || :
+    rm -f $netconf
+}
+
+function reset_bmc() {
+    local reset=${1:-'cold'}
+    ipmitool mc reset "$reset"
+    sleep 5 # Allow the BMC to go offline to prevent false-positive for connectivity checks.
+}
+
+function enable_amsd() {
+    if ! rpm -qi amsd >/dev/null 2>&1 ; then
+        echo 'amsd is not installed, ignoring amsd services'
+        return 0
+    fi
+    echo scanning vendor ... && vendor=$(ipmitool fru | grep -i 'board mfg' | tail -n 1 | cut -d ':' -f2 | tr -d ' ')
+    case $vendor in
+        *Marvell*|HP|HPE)
+            echo Enabling iLO services for detected vendor: $vendor
+            systemctl enable ahslog
+            systemctl enable amsd
+            systemctl enable smad
+
+            # Not needed; SCSI, IDE, nor FCA are used
```

Comment:
Why leave commented commands in the script?
```suggestion
            # Not needed; SCSI, IDE, nor FCA are used
```

Reply:
It pains me but I'm going to omit amending the metal-lib.sh script with changes to keep the diff to the original script in 1.3/1.4 the same.